DistanceScraper/Scrapers/BaseScraper.cs

```diff
@@ -184,36 +183,49 @@ public async Task RequestUserInfo(List<SteamID> steamIDs)
 				return;
 			}
 
+			Utils.WriteLine($"Requesting names for {steamIDs.Count} players");
 			IsRequestingUsers = new TaskCompletionSource<bool>();
-			steamIDs.ForEach(x => UsersToRequest.Enqueue(x));
-			var next = UsersToRequest.Dequeue();
-			Utils.WriteLine($"Requesting name for {next}");
-			RequestedSteamID = next;
-			Handlers.Friends.RequestFriendInfo(next, EClientPersonaStateFlag.PlayerName);
+			RequestedSteamIDs.AddRange(steamIDs);
+			foreach(var steamID in steamIDs)
+			{
+				if (Settings.Verbose)
+				{
+					Utils.WriteLine($"Requesting name for {steamID}");
+				}
+			}
+			Handlers.Friends.RequestFriendInfo(steamIDs, EClientPersonaStateFlag.PlayerName);
 
 			await IsRequestingUsers.Task;
 		}
 
 		public void OnRequestUserInfo(PersonaStateCallback callback)
 		{
+			if (Settings.Verbose)
+			{
+				Utils.WriteLine($"RequestedSteamIDs: {RequestedSteamIDs.Count}");
+			}
+			
 			// Skip callbacks from stuff we don't care about
-			if(callback.FriendID != RequestedSteamID)
+			if(!RequestedSteamIDs.Contains(callback.FriendID))
 			{
+				if (Settings.Verbose)
+				{
+					Utils.WriteLine($"Unexpected SteamID: {callback.FriendID}");
+				}
 				return;
 			}
```

Comment:
This helps with restarts of the scraper. Due to the way SteamKit and the steam game API works, responses from calls from the previous process can still be pending, and their response handlers will be triggered at the startup of the new process.